src/libraries/System.Data.Common/src/System/Data/DataRelationCollection.cs

```diff
@@ -336,7 +336,7 @@ internal int InternalIndexOf(string? name)
             if ((null != name) && (0 < name.Length))
             {
                 int count = List.Count;
-                int result = 0;
+                int result;
```

Comment:
This can be declared in the loop where it's assigned.

Reply:
Done